engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -4,55 +4,91 @@ import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.streaming.api.TimeCharacteristic
 import org.apache.flink.streaming.api.functions.source.SourceFunction
 import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
-import pl.touk.nussknacker.engine.api.MethodToInvoke
-import pl.touk.nussknacker.engine.api.process.{Source, SourceFactory}
+import pl.touk.nussknacker.engine.api.{Context, MethodToInvoke}
+import pl.touk.nussknacker.engine.api.process.{Source, SourceFactory, SourceTestSupport}
 import pl.touk.nussknacker.engine.flink.api.compat.ExplicitUidInOperatorsSupport
 import pl.touk.nussknacker.engine.flink.api.timestampwatermark.TimestampWatermarkHandler
+import pl.touk.nussknacker.engine.flink.util.context.{BasicFlinkContextInitializer, FlinkContextInitializer}
 
 import scala.reflect._
 
 /**
   * Source with methods specific for Flink
+  *
   * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
   */
+//TODO: remove type T parameter
 trait FlinkSource[T] extends Source[T] {
 
   def sourceStream(env: StreamExecutionEnvironment,
-                   flinkNodeContext: FlinkCustomNodeContext): DataStream[T]
+                   flinkNodeContext: FlinkCustomNodeContext): DataStream[Context]
+
+}
+
+/**
+  * Source with typical source stream trasformations:
+  *  - adds source using provided SourceFunction (here raw source produces raw data)
+  *  - sets UID
+  *  - assigns timestamp and watermarks
+  *  - initializes Context that is streamed within output DataStream (here raw data are mapped to Context)
+  * It separates raw event data produced by SourceFunction and data released to the stream as Context variables.
+  * By default it uses simple implementation of initializer, see [[pl.touk.nussknacker.engine.flink.util.context.BasicFlinkContextInitializer]].
+  *
+  * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
```

Comment:
Changed T -> Raw and fixed description.

Reply:
Sorry for being not precise, by "this description is correct?" I was meant the only "This is needed to handle e.g. syntax suggestions in UI" part. You removed it so it wasn't correct, right? (I'm asking because I don't know for sure). Thanks for very precise description BTW.

Reply:
I've reverted this part about syntax suggestions, it is used like that for sources with explicit MethodToInvoke annotation.